x-pack/plugins/maps/public/layers/styles/abstract_style.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ReactElement } from 'react';
+import { StyleDescriptor } from '../../../common/descriptor_types';
+import { ILayer } from '../layer';
+
+// todo: not sure if this is the right one
+export interface IStyle {
+  getDescriptor(): StyleDescriptor;
+  getDescriptorWithMissingStylePropsRemoved(): unknown;
+  pluckStyleMetaFromSourceDataRequest(): unknown;
+  renderEditor({
```

Comment:
This has actually been renamed to `renderStyleEditor`, https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/maps/public/connected_components/layer_panel/style_settings/style_settings.js#L14 and only takes onStyleDescriptorChange as a parameter

Reply:
Are you referring to `Layer#renderStyleEditor`? https://github.com/elastic/kibana/blob/37668b6da6dacf9e74f91c9e05c198109b52f155/x-pack/plugins/maps/public/layers/layer.tsx#L77

If so, I think we still need `IStyle#renderEditor`